contracts/contracts/GrantRoundManager.sol

```diff
@@ -99,66 +126,97 @@ contract GrantRoundManager {
   }
 
   /**
-   * @notice Swap and donate to a grant
-   * @param  _donation Donation being made to a grant
+   * @notice Performs swaps if necessary and donates funds as specified
+   * @param _swaps Array of SwapSummary objects describing the swaps required
+   * @param _deadline Unix timestamp after which a swap will revert, i.e. swap must be executed before this
+   * @param _donations Array of donations to execute
+   * @dev `_deadline` is not part of the `_swaps` array since all swaps can use the same `_deadline` to save some gas
+   * @dev Caller must ensure the input tokens to the _swaps array are unique
+   * @dev Does not verify
    */
-  function swapAndDonate(Donation calldata _donation) external payable {
+  function donate(
+    SwapSummary[] calldata _swaps,
+    uint256 _deadline,
+    Donation[] calldata _donations
+  ) external payable {
     // --- Validation ---
```

Comment:
Split up in 4fef97a4c52eb265f8a4db4ef8b39e05dd5ea5c0

Reply:
For context on the function naming, I used the same convention as the `app` of naming things `<parent-purpose>` to make it easier to understand something's purpose / where it fits in. So here the parent method is `donate` and we have 3 sub-methods of validation, swaps, and execution. I don't feel strongly about the naming, so I have no objections to renaming them if that reasoning doesn't convince you

Re: modifier, I'd personally vote against that since it's only used one time, so I'm not sure pulling it out adds any value, and just costs extra gas

Reply:
Though I do think we should add underscores to prefix internal methods regardless, but will hold off pending a final decision on naming

Reply:
Wasn't aware we had that convention on the frontend. Does that come from Vue?

Reply:
Yea it's based off their style guide: https://vuejs.org/v2/style-guide/#Tightly-coupled-component-names-strongly-recommended

Reply:
Got it. I can see how that convention makes sense for Components, but not sure it fits well for helper methods. The point of breaking the original function up was for code clarity, so easily grokable names seems pretty important. That said, don't feel strongly enough to block the PR or anything! I think I've made my case and will let you exercise your judgement!